sbin/prepareWorkspace.sh

```diff
@@ -452,25 +451,35 @@ checkingAndDownloadingFreeType() {
   if [[ -n "$FOUND_FREETYPE" ]]; then
     echo "Skipping FreeType download"
   else
-    downloadFile "freetype.tar.gz" "https://ci.adoptopenjdk.net/userContent/freetype/freetype-${BUILD_CONFIG[FREETYPE_FONT_VERSION]}.tar.gz"
-    downloadFile "freetype.tar.gz.sig" "https://ci.adoptopenjdk.net/userContent/freetype/freetype-${BUILD_CONFIG[FREETYPE_FONT_VERSION]}.tar.gz.sig"
-    checkFingerprint "freetype.tar.gz.sig" "freetype.tar.gz" "freetype" "58E0 C111 E39F 5408 C5D3 EC76 C1A6 0EAC E707 FDA5" "${FREETYPE_LIB_CHECKSUM}"
-
-    FREETYPE_BUILD_INFO="https://ci.adoptopenjdk.net/userContent/freetype/freetype-${BUILD_CONFIG[FREETYPE_FONT_VERSION]}.tar.gz"
-
+    # Delete existing freetype folder if it exists
     rm -rf "./freetype" || true
-    mkdir -p "freetype" || true
-    tar xpzf freetype.tar.gz --strip-components=1 -C "freetype"
-    rm freetype.tar.gz
+
+    case ${BUILD_CONFIG[FREETYPE_FONT_VERSION]} in
+    *.*)
+      # Replace . with - in version number e.g 2.8.1 -> 2-8-1
+      FREETYPE_BRANCH="VER-${BUILD_CONFIG[FREETYPE_FONT_VERSION]//./-}"
+      git clone https://github.com/freetype/freetype.git -b "${FREETYPE_BRANCH}" freetype || exit
+      ;;
+    *)
+      # Use specific git hash
+      git clone https://github.com/freetype/freetype.git freetype || exit
+      cd freetype || exit
+      git checkout "${BUILD_CONFIG[FREETYPE_FONT_VERSION]}" || exit
+      cd .. || exit
+      ;;
+    esac
+  
+    # Fetch the sha for the commit we just cloned
+    cd freetype || exit
+    FREETYPE_SHA=$(git rev-parse HEAD) || exit
+    FREETYPE_BUILD_INFO="https://github.com/freetype/freetype/commit/${FREETYPE_SHA}"
```

Comment:
We get the sha but we don't check it?  Or am I mistaken?

Reply:
the SHA is fetched for the SBOM (we don't need to check it IMO as I trust git to checkout the SHA we passed it originally)